Merge into main

damski-ugodnik opened the pull request.

Commit:
commit for PR

SergeyUdaltsov commented on controllers/processors/ProcessorFactory.java:
```diff
@@ -0,0 +1,35 @@
+import java.util.stream.Collectors;
+
+public class ProcessorFactory {
+    private final InputSpecification specification;
```
I am not sure this field is needed here

damski-ugodnik replied:
this one is used to pass the message for input

damski-ugodnik replied:
i will think about it

SergeyUdaltsov commented on controllers/processors/impl/BuyCoffeeProcessor.java:
```diff
@@ -0,0 +1,55 @@
+        if(title.equals("back")){
+            return;
+        }
+        RecipeProvider coffee = RecipeProvider.fromTitle(title);
```
please rename it to recipeProvider

SergeyUdaltsov commented on controllers/processors/impl/BuyCoffeeProcessor.java:
```diff
@@ -0,0 +1,55 @@
+
+    @Override
+    public void execute(Machine coffeeMachine) {
+        String title = ConsoleReader.readCommands(Constants.PURCHASE_SPECIFICATION);
```
please rename it to recipeTitle

SergeyUdaltsov commented on controllers/cookers/ICoffeeCooker.java:
```diff
@@ -0,0 +1,8 @@
+import models.CoffeeRecipe;
+import models.Machine;
+
+public interface ICoffeeCooker {
```
I think it is redundant here.

damski-ugodnik replied:
I use this to make cookers more extensible, so that if we add another fields, we can add fillers and cookers, which modify specific field

SergeyUdaltsov commented on models/RecipeProvider.java:
```diff
@@ -0,0 +1,56 @@
+        return str.toString();
+    }
+
+    RecipeProvider(String title) {
```
please move constructor up to the fields

damski-ugodnik replied:
ok, I'll fix it

SergeyUdaltsov commented on controllers/processors/ProcessorFactory.java:
```diff
@@ -0,0 +1,35 @@
+
+import specifications.InputSpecification;
+
+import java.util.*;
```
please do not use * imports

damski-ugodnik replied:
ok, got it

SergeyUdaltsov commented on models/RecipeProvider.java:
```diff
@@ -0,0 +1,56 @@
+    ESPRESSO("1") {
+        @Override
+        public CoffeeRecipe getRecipe() {
+            return new CoffeeRecipe(this.name().toLowerCase(), 250, 0, 16, 4);
```
do you need the title of recipe?

damski-ugodnik replied:
figured out, that this title is not used anywhere, tomorrow i will refactor this

damski-ugodnik committed:
removed unused CoffeeRecipe title;
moved RecipeProvider constructor up to fields;
replaced * imports with explicit ones;
renamed variables

damski-ugodnik committed:
removed non-code files

SergeyUdaltsov commented on CoffeeMachine.java:
```diff
@@ -0,0 +1,70 @@
+
+public class CoffeeMachine {
+
+
```
Please remove extra line space

SergeyUdaltsov commented on controllers/fillers/impl/CoffeeBeansFiller.java:
```diff
@@ -0,0 +1,18 @@
+public class CoffeeBeansFiller implements ICoffeeMachineFiller {
+    private final IntegerInputSpecification specification = new IntegerInputSpecification(
+            "Write how many grams of coffee beans you want to add:", "Invalid number",
+            Integer.MAX_VALUE, 0);
```
Please use more real maximum value here

SergeyUdaltsov commented on controllers/fillers/impl/CupsFiller.java:
```diff
@@ -0,0 +1,18 @@
+public class CupsFiller implements ICoffeeMachineFiller {
+    private final IntegerInputSpecification specification = new IntegerInputSpecification(
+            "Write how many disposable cups you want to add:", "Invalid number",
+            Integer.MAX_VALUE, 0);
```
the same here

SergeyUdaltsov commented on controllers/processors/ProcessorFactory.java:
```diff
@@ -0,0 +1,35 @@
+
+        for (ICoffeeMachineProcessor coffeeMachineProcessor : processorsList) {
+            operationNames.add(coffeeMachineProcessor.getOperationTitle());
+        }
```
you do not need this, because you already have operation titles in map. Try to use processors.keySet()

SergeyUdaltsov commented on controllers/processors/ProcessorFactory.java:
```diff
@@ -0,0 +1,35 @@
+import java.util.stream.Collectors;
+
+public class ProcessorFactory {
+    private final InputSpecification specification;
```
I am not sure that this is the correct place for the specification

SergeyUdaltsov commented on models/Machine.java:
```diff
@@ -0,0 +1,95 @@
+    private int money = 550;
+
+    public Machine() {
+        this.isOn = true;
```
please initialize all the fields in the same manner - in constructor or explicitly in fields

SergeyUdaltsov commented on models/Machine.java:
```diff
@@ -0,0 +1,95 @@
+
+    public void work(String command){
+
+    }
```
why do you need the empty method?

SergeyUdaltsov commented on models/Machine.java:
```diff
@@ -0,0 +1,95 @@
+    }
+
+    public void setOn(boolean on) {
+        isOn = on;
```
please rename the argument on -> isOn

SergeyUdaltsov commented on models/RecipeProvider.java:
```diff
@@ -0,0 +1,57 @@
+        }
+    };
+
+
```
please remove extra empty line

SergeyUdaltsov commented on models/RecipeProvider.java:
```diff
@@ -0,0 +1,57 @@
+        return str.toString();
+    }
+
+
```
the same

SergeyUdaltsov commented on models/RecipeProvider.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    public static RecipeProvider fromTitle(String title) {
+
```
the same

SergeyUdaltsov commented on utils/ConsoleReader.java:
```diff
@@ -0,0 +1,35 @@
+            System.out.println(specification.inputMessage());
+            input = scanner.next();
+            if (InputValidator.validateInt(input, specification)) {
+                return Integer.parseInt(input);
```
you parse the input two times. Please rework this logic to avoid extra operations

damski-ugodnik committed:
code refactoring
